### PR TITLE
ConsisID for xdit

### PR DESCRIPTION
**Pull request by pkuhxy**
I'm sorry that my misoperation caused the previous repository to become invalid, so I resubmitted a pull request.

### CODE REVIEW

**Comment by pkuhxy** [comment]
Page: https://pku-yuangroup.github.io/ConsisID/
Code: https://github.com/PKU-YuanGroup/ConsisID
Demo: https://huggingface.co/spaces/BestWishYsh/ConsisID-preview-Space

File: examples/run_cogvideo.sh [hunk 1]
```diff
@@ -5,7 +5,7 @@ export PYTHONPATH=$PWD:$PYTHONPATH
 
 # CogVideoX configuration
 SCRIPT="cogvideox_example.py"
-MODEL_ID="/cfs/dit/CogVideoX1.5-5B"
+MODEL_ID="/storage/ysh/Ckpts/CogVideoX1.5-5B"
```
Comment (feifeibear) [inline examples/run_cogvideo.sh]: could you please revert this file back?

File: examples/run_consisid.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+PARALLEL_ARGS="--ulysses_degree 2 --ring_degree 3"
+# CFG_ARGS="--use_cfg_parallel"
+
+# Uncomment and modify these as needed
+# PIPEFUSION_ARGS="--num_pipeline_patch 8"
+# OUTPUT_ARGS="--output_type latent"
+# PARALLLEL_VAE="--use_parallel_vae"
+ENABLE_TILING="--enable_tiling"
+# COMPILE_FLAG="--use_torch_compile"
+INPUT_IMAGE='--image_path /storage/hxy/cvpr2025/ConsisID/asserts/example_images/4.png'
```
Comment (feifeibear) [inline examples/run_consisid.sh]: Could you please use a relative path here and give instruction to users how to download this image?

File: xfuser/core/distributed/runtime_state.py [hunk 3]
```diff
@@ -103,7 +103,11 @@ def __init__(self, pipeline: DiffusionPipeline, config: EngineConfig):
             pipeline=pipeline, parallel_config=config.parallel_config
         )
         self.cogvideox = False
-        if isinstance(pipeline, CogVideoXPipeline):
+        if isinstance(pipeline, CogVideoXPipeline) or isinstance(pipeline, ConsisIDPipeline):
+
+            # import pdb; pdb.set_trace()
+            print("consisid get in this set_cogvideox_parameters")
```
Comment (feifeibear) [inline xfuser/core/distributed/runtime_state.py]: can we remove the debug code?

File: xfuser/core/distributed/runtime_state.py [hunk 4]
```diff
@@ -112,6 +116,9 @@ def __init__(self, pipeline: DiffusionPipeline, config: EngineConfig):
                 backbone_inner_dim=pipeline.transformer.config.num_attention_heads
                 * pipeline.transformer.config.attention_head_dim,
             )
+
+            print(f"self.cogvideox:{self.cogvideox}")
```
Comment (feifeibear) [inline xfuser/core/distributed/runtime_state.py]: can we remove the debug code?

File: xfuser/model_executor/layers/attention_processor.py [hunk 5]
```diff
@@ -1143,3 +1143,176 @@ def __call__(
             [text_seq_length, latent_seq_length], dim=1
         )
         return hidden_states, encoder_hidden_states
+
+
+@xFuserAttentionProcessorRegister.register(CogVideoXAttnProcessor2_0)
+class xFuserConsisIDAttnProcessor2_0(CogVideoXAttnProcessor2_0):
+    r"""
+    Processor for implementing scaled dot-product attention for the CogVideoX model. It applies a rotary embedding on
```
Comment (feifeibear) [inline xfuser/model_executor/layers/attention_processor.py]: for the ConsisID model

**Review by feifeibear — CHANGES_REQUESTED** [review]

**Comment by feifeibear** [comment]
You can also update the readme page with consistid on news and supported model session.

**Comment by feifeibear** [comment]
@pkuhxy Xianyi, the PR looks perfect!  
I assume you have nailed the PipeFusion + USP for consistid. 
If you can paste a screenshot demonstrating the consistent run with hybrid parallel successfully, I will proceed with merging the PR.

**Comment by SHYuanBest** [comment]
I have fix some bugs and reorganize the code. to do:

- [ ] Update main README.md
- [ ] Add Performance README.md

input: https://github.com/PKU-YuanGroup/ConsisID/blob/main/asserts/example_images/2.png?raw=true

https://github.com/user-attachments/assets/ecf2bced-46dc-4819-9245-6110b63dea17

input: https://github.com/PKU-YuanGroup/ConsisID/blob/main/asserts/example_images/1.png?raw=true

https://github.com/user-attachments/assets/25525796-49a3-4815-8e29-3428c632e36c

**Comment by pkuhxy** [comment]
> @pkuhxy Xianyi, the PR looks perfect! I assume you have nailed the PipeFusion + USP for consistid. If you can paste a screenshot demonstrating the consistent run with hybrid parallel successfully, I will proceed with merging the PR.

We have completed all parts of this PR. Please review the code and merge it. Thank you!

File: examples/run_consisid.sh [hunk 6]
```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
```
Comment (feifeibear) [inline examples/run_consisid.sh:1]: dose this script work with pipefusion degree>1?

File: examples/consisid_example.py [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+def main():
+    parser = FlexibleArgumentParser(description="xFuser Arguments")
+    args = xFuserArgs.add_cli_args(parser).parse_args()
+    engine_args = xFuserArgs.from_cli_args(args)
+
+    engine_config, input_config = engine_args.create_config()
+    local_rank = get_world_group().local_rank
+
+    assert engine_args.use_parallel_vae is False, "parallel VAE not implemented for ConsisID"
+
```
Comment (feifeibear) [inline examples/consisid_example.py:30]: if you have not implemented pipefusion yet, you can raise an error when setting pipefuison > 1

**Review by feifeibear — CHANGES_REQUESTED** [review]

File: examples/consisid_example.py (hunk 7 above)
Comment (SHYuanBest) [reply examples/consisid_example.py:30]: @feifeibear Have update. Thanks for your review.

**Review by feifeibear — APPROVED** [review]

**Pull request merged**